curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/5d5a813321b9e3db6c106a46.md

```diff
@@ -119,6 +119,23 @@ const title = document.querySelector('title');
 assert.exists(title);
 ```
 
+
+
+You should have an opening `title` tag.
+
+```js
+assert(code.match(/<title\s*>/i));
+```
```

Comment:
I’ll also note that this test is more restrictive than the above one because it would fail for `<title id="foo">`, which would be an odd choice, but a syntactically correct one (and it would then be confusing to be alerted that you’re “missing” the opening tag when you clearly are not.

Reply:
> This appears to be a duplicate of the "Your code should have a `title` element." above. Is there a scenario where the users pass the above case but fail this one?

@huyenltnguyen Yes you are right, checking for opening `<title>` tag, is unnecessary since we are already checking the existence of `title` tag. I'll make the adjustments.

Reply:
> I’ll also note that this test is more restrictive than the above one because it would fail for `<title id="foo">`, which would be an odd choice, but a syntactically correct one. And it would then be confusing to be alerted that you’re “missing” the opening tag when you clearly are not.

@plamoni  yes, you are right, I guess this PR is lacking obvious testing on my end, I'll make the necessary changes.